From e9b8cd9dcd39f6f864eea2e12856e65cb3a82442 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Fri, 10 May 2024 08:34:29 +0200
Subject: [PATCH] sched.h: Fix compile error with HMP scheduler used in the MSM
 4.9 kernels (tissot, mido, rosy, sargo)

HMP scheduler which seems MSM kernel specific was introduced in the Mido, Tissot, Rosy and Sargo kernels with https://github.com/shr-distribution/linux/commit/dddcab78bdda9d63b5cf6ddcdc30f815d98e8d98 or similar commit such as https://github.com/shr-distribution/linux/commit/c662231a6bfe77e202254913f9c51d9377650e8f. However HMP support never made it in this way to mainline kernel in this way and lttng-modules doesn't accommodate for it.

Simply patch it in here.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Inappropriate [LuneOS Specific]

 include/instrumentation/events/sched.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/instrumentation/events/sched.h b/include/instrumentation/events/sched.h
index 24cf37c8..ef7ac2f6 100644
--- a/include/instrumentation/events/sched.h
+++ b/include/instrumentation/events/sched.h
@@ -434,14 +434,15 @@ LTTNG_TRACEPOINT_EVENT(sched_switch,
  */
 LTTNG_TRACEPOINT_EVENT(sched_migrate_task,
 
-	TP_PROTO(struct task_struct *p, int dest_cpu),
+	TP_PROTO(struct task_struct *p, int dest_cpu, unsigned int load),
 
-	TP_ARGS(p, dest_cpu),
+	TP_ARGS(p, dest_cpu, load),
 
 	TP_FIELDS(
 		ctf_array_text(char, comm, p->comm, TASK_COMM_LEN)
 		ctf_integer(pid_t, tid, p->pid)
 		ctf_integer(int, prio, p->prio - MAX_RT_PRIO)
+		ctf_integer(unsigned int, load, load )
 		ctf_integer(int, orig_cpu, task_cpu(p))
 		ctf_integer(int, dest_cpu, dest_cpu)
 	)
